Spigot-Server-Patches/0565-Optimize-Villagers.patch

```diff
@@ -70,19 +79,18 @@ index 35eb3a5a61145e94d5b0c77c0eb13bfa46fac23b..6861b1a345496a83900b0ef702ba3431
 +
 +                        BlockPosition pos = record.getPosition();
 +                        long key = pos.asLong();
-+                        if (this.e.containsKey(key)) {
++                        if (this.g.containsKey(key)) {
 +                            continue;
 +                        }
 +                        double poiDist = pos.distanceSquared(blockposition2);
 +                        if (poiDist <= (double) requiredDist) {
-+                            this.e.put(key, (long) (this.d + Math.sqrt(poiDist) * 4)); // use dist instead of 40 to blacklist longer if farther distance
++                            this.g.put(key, new BehaviorFindPosition.a(java.util.concurrent.ThreadLocalRandom.current(), (long) (this.f + Math.sqrt(poiDist) * 4))); // use dist instead of 40 to blacklist longer if farther distance
 +                            ++poiAttempts;
-+                            PathEntity pathentity = entitycreature.getNavigation().a(com.google.common.collect.ImmutableSet.of(pos), 8, false, this.a.d());
++                            PathEntity pathentity = entitycreature.getNavigation().a(com.google.common.collect.ImmutableSet.of(pos), 8, false, this.b.getValidRange());
 +
-+                            if (pathentity != null && pathentity.h()) {
-+                                record.decreaseVacancy();
```

Comment:
Where'd the `decreaseVacancy` call go? We still need to make it clear that it's *not* available, do we not?

Reply:
![image](https://img.mrivanplays.com/msedge_HYKypQGkUS.png)